Cubical/Categories/Constructions/Hyperdoctrine.agda

```diff
@@ -0,0 +1,63 @@
+{-# OPTIONS --safe #-}
+
+module Cubical.Categories.Constructions.Hyperdoctrine where
```

Comment:
I think this module shouldn't be `Hyperdoctrine` but instead something like `Hyperdoctrine.Unary` or `UnaryOverCartesian` because we will have other notions of hyperdoctrine as well.

In fact there are two axes of generalization: what structure we assume on the base category and what structure we assume about the fibers.

There's an even more basic version than the one defined here, which is that we could drop the requirement that the base category has products.

Reply:
Created a Hyperdoctrine directory and named this file Unary.agda.